### PR TITLE
CodeGen: Fix libcall names for exp10 on the various darwins

### PR DESCRIPTION
**Pull request by arsenm**
It's really great that we have the same information duplicated in TargetLibraryInfo and RuntimeLibcalls which both assume everything by default.

Should fix issue reported after #92287

### CODE REVIEW

File: llvm/lib/CodeGen/TargetLoweringBase.cpp [hunk 1]
```diff
@@ -227,6 +227,34 @@ void TargetLoweringBase::InitLibcalls(const Triple &TT) {
...
+        setLibcallName(RTLIB::EXP10_F32, "__exp10f");
+        setLibcallName(RTLIB::EXP10_F64, "__exp10");
+      }
+      break;
+    case Triple::IOS:
+    case Triple::TvOS:
+    case Triple::WatchOS:
+    case Triple::XROS:
+      if (!TT.isWatchOS() &&
+          (TT.isOSVersionLT(7, 0) || (TT.isOSVersionLT(9, 0) && TT.isX86()))) {
```
Comment (fhahn) [inline llvm/lib/CodeGen/TargetLoweringBase.cpp:246]: Can those checks be simplified? Looking at `math.h`, it looks like they should be available on iOS from 7.0 and for any version of tvOS, watchOS and xrOS?

Comment (arsenm) [reply llvm/lib/CodeGen/TargetLoweringBase.cpp:246]: These are copied verbatim from what TargetLibraryInfo does; I'm not really looking to improve on this here now

Comment (fhahn) [reply llvm/lib/CodeGen/TargetLoweringBase.cpp:246]: Oh right, makes sense

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**